pystac_client/cli.py

```diff
@@ -368,12 +358,7 @@ def cli() -> int:
             return search(client, **args)
         elif cmd == "collections":
             return collections(client, **args)
-        else:
```

Comment:
I'm ok removing the branch, but personally I don't hate having it, as a protection in case we add or remove subcommands at some point in the future?

If we wanted to make it clear that it's unreachable, could change the "nice" error handling to a:

```python
assert(False)  # Branch should be unreachable
```

Reply:
yeah I don't hate having it either. My only worry with that kind of thing is that the error handling might be broken and we wouldn't notice or it gives us a false sense of security. I will add it back in though. I think I agree that it is better to try to handle it.